### PR TITLE
[USM] add TLS to protocol stack if TLS static tags are present

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

This PR adds TLS to the protocol stack if one of the TLS static tags are present.

### Describe how to test/QA your changes

Change is covered by automated tests.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/encoding/format_usm_test.go [hunk 1]
```diff
@@ -92,10 +93,43 @@ func TestFormatProtocols(t *testing.T) {
...
+		{
+			name:       "GoTLS - unknown protocol",
+			protocol:   network.ProtocolUnknown,
+			staticTags: 1 << 2,
+			want: &model.ProtocolStack{
+				Stack: []model.ProtocolType{
+					model.ProtocolType_protocolTLS,
+					model.ProtocolType_protocolUnknown,
+				},
+			},
```
Comment (guyarb) [inline pkg/network/encoding/format_usm_test.go:127]: 1. let's use the consts
2. let's mix the tests, one test with unknown, one with http and the left with db protocol
3. let's add a test for unsupport tag (assume `1 << 10`)

File: pkg/network/encoding/format.go [hunk 2]
```diff
@@ -295,14 +297,18 @@ func unsafeStringSlice(key string) []byte {
 //				model.ProtocolType_protocolHTTP2,
 //			},
 //		}
-func formatProtocol(protocol network.ProtocolType) *model.ProtocolStack {
+func formatProtocol(protocol network.ProtocolType, staticTags uint64) *model.ProtocolStack {
+	stack := make([]model.ProtocolType, 0)
+	if staticTags&tlsTagsMask > 0 {
```
Comment (guyarb) [inline pkg/network/encoding/format.go:302]: i think we should create a function `IsTLSTag` next to `GetStaticTag`
and check it there. furthermore, add a comment next to the definitions of the TLS tags

File: pkg/network/encoding/format.go [hunk 3]
```diff
@@ -295,14 +297,18 @@ func unsafeStringSlice(key string) []byte {
 //				model.ProtocolType_protocolHTTP2,
 //			},
 //		}
-func formatProtocol(protocol network.ProtocolType) *model.ProtocolStack {
+func formatProtocol(protocol network.ProtocolType, staticTags uint64) *model.ProtocolStack {
+	stack := make([]model.ProtocolType, 0)
```
Comment (guyarb) [inline pkg/network/encoding/format.go:301]:
```suggestion
	stack := make([]model.ProtocolType, 0, 1)
```

File: pkg/network/encoding/format.go [hunk 4]
```diff
@@ -295,14 +297,18 @@ func unsafeStringSlice(key string) []byte {
 //				model.ProtocolType_protocolHTTP2,
 //			},
 //		}
-func formatProtocol(protocol network.ProtocolType) *model.ProtocolStack {
+func formatProtocol(protocol network.ProtocolType, staticTags uint64) *model.ProtocolStack {
```
Comment (guyarb) [inline pkg/network/encoding/format.go:300]: modify the documentation above

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/tags_linux.go [hunk 5]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/network/protocols/http"
 )
 
+const tlsTagsMask = http.GnuTLS | http.OpenSSL | http.Go
```
Comment (guyarb) [inline pkg/network/tags_linux.go:15]: consider adding a comment to `http_types.go` a comment above the definition of the tags, so people will add here their new tags, or move the definition of `tlsTagMask` to `http_types.go`

**Review by guyarb — APPROVED** [review]
Looks good, leaving a suggestion for improvement

**Review by hmahmood — APPROVED** [review]

**Pull request merged**